server/controller.go

```diff
@@ -135,94 +137,46 @@ func getProfileData(r *http.Request) (map[string]string, error) {
 	resp, err := client.Do(req)
 	if err != nil {
 		return m, err
 	}
 	defer resp.Body.Close()
 
-	body, err := ioutil.ReadAll(resp.Body)
+	body, err := io.ReadAll(resp.Body)
 	if err != nil {
 		return m, err
 	}
 
-	json.Unmarshal(body, &m)
-
-	return m, nil
-}
-
-func AuthCodeCallbackHandler(c *gin.Context) {
-	session, err := sessionStore.Get(c.Request, "okta-hosted-login-session-store")
+	err = json.Unmarshal(body, &m)
 	if err != nil {
-		c.AbortWithError(http.StatusForbidden, err)
-		return
-	}
-
-	// Check the state that was returned in the query string is the same as the above state
-	if c.Query("state") == "" || c.Query("state") != session.Values["oauth_state"] {
-		c.AbortWithError(http.StatusForbidden, fmt.Errorf("the state was not as expected"))
-		return
-	}
-
-	// Make sure the code was provided
-	if c.Query("error") != "" {
-		c.AbortWithError(http.StatusForbidden, fmt.Errorf("authorization server returned an error: %s", c.Query("error")))
-		return
-	}
-
-	// Make sure the code was provided
-	if c.Query("code") == "" {
-		c.AbortWithError(http.StatusForbidden, fmt.Errorf("the code was not returned or is not accessible"))
-		return
-	}
-
-	token, err := oktaOauthConfig.Exchange(
-		context.Background(),
-		c.Query("code"),
-		oauth2.SetAuthURLParam("code_verifier", session.Values["oauth_code_verifier"].(string)),
-	)
-	if err != nil {
-		c.AbortWithError(http.StatusUnauthorized, err)
-		return
-	}
-
-	// Extract the ID Token from OAuth2 token.
-	rawIDToken, ok := token.Extra("id_token").(string)
-	if !ok {
-		c.AbortWithError(http.StatusUnauthorized, fmt.Errorf("id token missing from OAuth2 token"))
-		return
-	}
-	_, err = verifyToken(rawIDToken)
-
-	if err != nil {
-		c.AbortWithError(http.StatusForbidden, err)
-		return
-	} else {
-		session.Values["access_token"] = token.AccessToken
-
-		session.Save(c.Request, c.Writer)
+		return m, err
 	}
 
-	c.Redirect(http.StatusFound, "/")
+	return m, nil
 }
 
-func verifyToken(t string) (*verifier.Jwt, error) {
-	tv := map[string]string{}
-	tv["aud"] = os.Getenv("OKTA_OAUTH2_CLIENT_ID")
-	jv := verifier.JwtVerifier{
-		Issuer:           os.Getenv("OKTA_OAUTH2_ISSUER"),
-		ClaimsToValidate: tv,
-	}
-
-	result, err := jv.New().VerifyIdToken(t)
-	if err != nil {
-		return nil, fmt.Errorf("%s", err)
-	}
-
-	if result != nil {
-		return result, nil
-	}
-
-	return nil, fmt.Errorf("token could not be verified")
+func AuthCodeCallbackHandler(c *gin.Context) {
+	// Use zitadel's built-in callback handler which properly handles PKCE cookies
+	marshalToken := func(w http.ResponseWriter, r *http.Request, tokens *oidc.Tokens[*oidc.IDTokenClaims], state string, rp rp.RelyingParty) {
+		// Store access token in session
+		session, err := sessionStore.Get(r, "okta-hosted-login-session-store")
+		if err != nil {
+			http.Error(w, "Failed to get session", http.StatusInternalServerError)
+			return
+		}
+
+		session.Values["access_token"] = tokens.AccessToken
```

Comment:
We really should be using the ID token, not access token. Access token is only used for making resource requests, in this case, Okta org resources.